.github/workflows/test.yml

```diff
@@ -0,0 +1,41 @@
+name: Test
+on: 
+  pull_request_target:
+    types: [assigned, opened, synchronize, reopened, ready_for_review]
+    paths: 
+        - libdash/**
+        - shasta/**
+  push:
+    branches:
+      - main
+      - future
+    paths: 
+      - libdash/**
+      - shasta/**
+jobs:
+  Shasta-Test:
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - ubuntu 20.04
+    runs-on: ${{ matrix.os }}
+    if: github.event.pull_request.draft == false
+    steps:
+      - uses: actions/checkout@v2 
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+      - name: Running Tests
+        run: |
+          sudo touch /.githubenv
+          (cd libdash; python setup.py install)
```

Comment:
This can go in a `setup_test.sh` script that also pulls libdash either from pip or github.